File: Program/1_Array/Array Standard problem/3_move_all_zeroes_to_end.c
#include <stdio.h>

void pushZerosToEnd(int arr[], int n)
{
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (arr[i] != 0)
        {
            arr[count] = arr[i];
            count++;
        }
    }

    while (count < n)
    {
        arr[count] = 0;
        count++;
    }
}

int main()
{
    int arr[] = {1, 9, 8, 4, 0, 0, 2, 7, 0, 6, 0, 9};
    int n = sizeof(arr) / sizeof(arr[0]);
    pushZerosToEnd(arr, n);
    printf("%s\n", "Array after pushing all zeros to the end of array:");
    for (int i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }
    return 0;
}

File: Program/1_Array/Array Standard problem/easy/3_move_all_zeroes_to_end.py
def move_all_zero(arr):
    count = 0
    for i in range(len(arr)):
        if arr[i] != 0:
            arr[count] = arr[i]
            count+=1
    
    while count < len(arr):
        arr[count] = 0
        count+=1
    return arr



l=[1, 9, 8, 4, 0, 0, 2, 7, 0, 6, 0, 9]
print(move_all_zero(l))


